imports/client/components/ChatMessageV2.tsx

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable react/no-array-index-key */
+import { marked } from 'marked';
+import React from 'react';
+import styled from 'styled-components';
+import { ChatMessageContentNodeType, ChatMessageContentType, ChatMessageMentionNodeType } from '../../lib/schemas/ChatMessage';
+import { MentionSpan } from './FancyEditor';
+
+// This file implements standalone rendering for the MessageElement format
+// defined by FancyEditor, for use in the chat pane.
+
+const StyledMessage = styled.p`
+  margin-bottom: 0;
+`;
+
+const PreWrapSpan = styled.span`
+  white-space: pre-wrap;
+`;
+
+const StyledBlockquote = styled.blockquote`
+  border-left: 2px solid #88a;
+  padding-left: 4px;
+  margin-bottom: 0;
+`;
+
+// For the readonly renderer
+const StyledCodeBlock = styled.code`
+  white-space: pre-wrap;
+  display: block;
+  border-radius: 4px;
+  padding: 4px;
+  width: 100%;
+  background-color: #eee;
+  color: black;
+  margin-bottom: 0;
+`;
+
+function isMention(thing: ChatMessageContentNodeType): thing is ChatMessageMentionNodeType {
+  return (thing as any).type === 'mention';
```

Comment:
(similarly this could phrase as `'type' in thing && thing.type === 'mention'`

Reply:
This can just use the shared function from the ChatMessage schema file.  Fixed in 7cbd867e0b78b131bd0536cd99367ba791475764